src/main/scala/mesosphere/marathon/core/event/impl/stream/HttpEventStreamServlet.scala

```diff
@@ -27,16 +27,17 @@ class HttpEventSSEHandle(request: HttpServletRequest, emitter: Emitter) extends
 
   private val subscribedEventTypes = request.getParameterMap.getOrDefault("event_type", Array.empty).toSet
 
-  def subscribed(eventType: String): Boolean = {
-    subscribedEventTypes.isEmpty || subscribedEventTypes.contains(eventType)
-  }
+  def subscribedForEventType(eventType: String): Boolean = subscribedEventTypes.isEmpty || subscribedEventTypes.contains(eventType)
+  def subscribedForTaskId(taskId: String): Boolean       = subscribedEventTypes.isEmpty || subscribedEventTypes.contains(task_id)
 
   override def remoteAddress: String = request.getRemoteAddr
 
   override def close(): Unit = emitter.close()
 
   override def sendEvent(event: MarathonEvent): Unit = {
-    if (subscribed(event.eventType)) blocking(emitter.event(event.eventType, event.jsonString))
+  	if subscribedForEventType(event.eventType)) && subscribedForTaskId(event.taskId)) blocking(emitter.event(event.eventType, event.jsonString))
```

Comment:
This line is redundant. Actually, as is, this function will be duplicating events.

Reply:
The idea was to make a case for filtering by taskId, filtering by eventType and case of filtering by both. I can't figure out how to return `blocking...` for both.